src/wp-includes/functions.php

```diff
@@ -6039,6 +6039,45 @@ function _doing_it_wrong( $function_name, $message, $version ) {
 	}
 }
 
+/**
+ * Generates a user-level error/warning/notice/deprecation message.
+ *
+ * Generates the message when `WP_DEBUG` is true.
+ *
+ * @since 6.4.0
+ *
+ * @param string $function_name The function that triggered the error.
+ * @param string $message       The message explaining the error.
+ * @param int    $error_level   Optional. The designated error type for this error.
+ *                              Only works with E_USER family of constants. Default E_USER_NOTICE.
+ */
+function wp_trigger_error( $function_name, $message, $error_level = E_USER_NOTICE ) {
+
+	// Bail out if WP_DEBUG is not turned on.
+	if ( ! WP_DEBUG ) {
+		return;
+	}
+
+	/**
+	 * Fires when the given function triggers a user-level error/warning/notice/deprecation message.
```

Comment:
It's not a filter, it's an action :) 

Actually wondering which may be more useful here:
- Seems an action can only be used to do `debug_backtrace()` (although there is a warning about using it in PHP 7+).
- A filter can also be used for `debug_backtrace()` but can also be used to short-circuit the rest of the function. Could that be useful in some cases?

Reply:
>A filter can also be used for debug_backtrace() but can also be used to short-circuit the rest of the function. Could that be useful in some cases?

What would be a valid use case to short-circuit `trigger_error()` when `WP_DEBUG` is on?

Reply:
Each of the `_deprecated_*()` and `_doing_it_wrong()` already have a short-circuit built into them. I'm thinking `wp_trigger_error()` doesn't need a short-circuit, as it's just a `trigger_error()` wrapper.